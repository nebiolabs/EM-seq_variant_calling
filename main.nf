#!/usr/bin/env nextflow
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    EM-seq Variant Calling Workflow
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
----------------------------------------------------------------------------------------
*/

nextflow.enable.dsl = 2

// To do maybe populate valid workflows list from workflows directory
// if (params.workflows && params.workflow && !params.workflows.containsKey(params.workflow)) {
//   exit 1, "The provided genome '${params.genome}' is not available in the genomes file. Currently the available genomes are ${params.genomes.keySet().join(", ")}"
// }

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    NAMED WORKFLOW FOR PIPELINE
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

include { emseq_variant_calling } from './workflows/emseq_variant_calling.nf'

//
// WORKFLOW: Run main analysis pipeline
//
workflow variant_calling {
    emseq_variant_calling()
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    RUN ALL WORKFLOWS
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//
// WORKFLOW: Execute a single named workflow for the pipeline
//
workflow {
    variant_calling ()
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    THE END
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/
