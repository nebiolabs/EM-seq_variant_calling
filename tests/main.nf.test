nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "pipeline"

    test("Pipeline test with small dataset") {

        when {
            params.fasta = "${projectDir}/tests/fixtures/genome/t2t_2.0_chr21.fa"
            params.fai = "${projectDir}/tests/fixtures/genome/t2t_2.0_chr21.fa.fai"
            params.bams = "${projectDir}/tests/fixtures/bams/200ng-V2-{1,2}.md.bam"
        }

        then {           

            def calcmd_bam = path("${launchDir}/test_output/calcMD/200ng-V2-1.md.calcmd.bam").md5
            def masked_bam = path("${launchDir}/test_output/revelio/200ng-V2-1.md.masked.bam").md5
            def strelka_vcf = path("${launchDir}/test_output/strelka/200ng-V2-1.md.strelka_vcf.gz").vcf.summary

            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.trace,
                    ["calcmd_bam", calcmd_bam],
                    ["masked_bam", masked_bam],
                    ["strelka_vcf", strelka_vcf]
                                        ).match()
                }
            )

        }

    }

}
