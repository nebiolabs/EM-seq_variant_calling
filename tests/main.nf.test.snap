{
    "Pipeline test with small dataset": {
        "content": [
            {
                "tasksFailed": 0,
                "tasksCount": 5,
                "tasksSucceeded": 5
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.2"
        },
        "timestamp": "2025-03-06T10:31:10.628849283"
    }
}